### PR TITLE
refactoring README and including architecture design

### PR DESCRIPTION
**Pull request by julioalex-rezende**
writing README to reflect recent changes to hte code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Java Workflow Sample using DAPR
```
Comment (macromania) [inline README.md]:
```suggestion
# Dapr Workflows for Java Sample
```

File: README.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
-Initial playgroud to work with DAPR workflows in Java
+Initial playgroud to work with DAPR workflows in Java in a distributed environment
```
Comment (macromania) [inline README.md]:
```suggestion
Initial playgroud to work with DAPR workflows in Java in k8s.
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
 
-This sample is a simple money transfer application built on Java Spring Boot, leveraging DAPR workflows to manage the creation of accounts and facilitate money transfers between accounts. The project focuses on simplicity, reliability, and extensibility, utilizing DAPR for state management and implementing various workflow patterns.
+This sample depicts a simple money transfer application built on Java Spring Boot, leveraging DAPR workflows, service invocation, and sidecar patterns to manage the creation of accounts facilitate money transfers between accounts running in a Kubernetes cluster. The project focuses on simplicity, reliability, and extensibility, utilizing DAPR for state management and implementing some of the workflow patterns described in DAPR Documentation in a distributed application.
```
Comment (macromania) [inline README.md]:
```suggestion
This sample depicts a simple money transfer application built on Java Spring Boot, leveraging Dapr Workflows for Java and  Dapr Service Invocation feature to manage account creation and money transfers in a k8s. The project focuses on simplicity, reliability, and extensibility, utilizing Dapr for state management and implementing some of [the workflow patterns](https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-patterns/) described in Dapr documentation.
```

File: README.md [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+
+It makes use of out-of-the-box capabilities of DAPR to manage workflows, handling requirements such as retries and durability without the need of extra features' implementation.
```
Comment (macromania) [inline README.md]:
```suggestion
It uses Dapr Workflows for Java for handling non-functional requirements such as retries and durability out of box.
```

File: README.md [hunk 5]
```diff
@@ -17,16 +19,47 @@ This sample is a simple money transfer application built on Java Spring Boot, le
...
+
+The project contains the following services:
+
+- [Public API](/src/public-api-service) - Public API endpoint that receives new money transfer and account creation requests and calls the respective service via DAPR Service Method Invokation, that starts workflow and perform the operations. Public API can also access the state-store directly to check for status
+- [Account Service](/src/account-service) - Service that handles the workflow for account creation.
+- [Transfer Service](/src/transfer-service) - Transfer service that handles the workflow for a money transfer request. 
+
+### How it works?
+
+1. Public API endpoint receives new requests for accounts or transfers.
```
Comment (macromania) [inline README.md]:
```suggestion
1. Public API endpoint receives new requests for account creation or money transfers. 
```

File: README.md [hunk 6]
```diff
@@ -17,16 +19,47 @@ This sample is a simple money transfer application built on Java Spring Boot, le
...
+The project contains the following services:
+
+- [Public API](/src/public-api-service) - Public API endpoint that receives new money transfer and account creation requests and calls the respective service via DAPR Service Method Invokation, that starts workflow and perform the operations. Public API can also access the state-store directly to check for status
+- [Account Service](/src/account-service) - Service that handles the workflow for account creation.
+- [Transfer Service](/src/transfer-service) - Transfer service that handles the workflow for a money transfer request. 
+
+### How it works?
+
+1. Public API endpoint receives new requests for accounts or transfers. 
+2. Public API invokes the correct service via DAPR Sidecars, depending on the request (account creation or money transfer).
```
Comment (macromania) [inline README.md]:
```suggestion
2. Public API invokes corresponding service depending on account creation or money transfer request using Service Invocation pattern and starts corresponding workflow in that service.
```

File: README.md [hunk 7]
```diff
@@ -17,16 +19,47 @@ This sample is a simple money transfer application built on Java Spring Boot, le
...
+    2.1. Account Creation Workflow Starts
+        2.1.1. Multiple tasks are performed via Workflow Activities called by the DAPR client such as Notification, Approval, Creation.
```
Comment (macromania) [inline README.md]:
```suggestion
        2.1.1. Multiple tasks are performed via Workflow Activities called by the Dapr client such as Notification, Approval, Creation.
```

File: README.md [hunk 8]
```diff
@@ -17,16 +19,47 @@ This sample is a simple money transfer application built on Java Spring Boot, le
...
+        2.1.2. A validation if the amount is valid for creating an account (greater than zero) is performed.
+    2.2. Transfer workflow starts
+        2.2.1. Multiple tasks are performed via Workflow Activities called by the DAPR client such as Notification, Fraud Detection, Transfer.
```
Comment (macromania) [inline README.md]:
```suggestion
        2.2.1. Multiple tasks are performed via Workflow Activities called by the Dapr client such as Notification, Fraud Detection, Transfer.
```

**Review by macromania — COMMENTED** [review]
Mostly rewording suggestions 😅

Good work 👏🏼

**Review by Aymalla — COMMENTED** [review]
I've a question, Do we have away to see the running workflow status history?

**Comment by julioalex-rezende** [comment]
> I've a question, Do we have away to see the running workflow status history?

AFAIK, it's possible to query the current status of a workflow instance through the methods exposed in the SDK. not sure about the history.
This sample doesn't do that though, it focuses on the accounts/transfers' status.

**Pull request merged**